src/components/partnership/partnership.service.ts

```diff
@@ -603,4 +627,71 @@ export class PartnershipService {
       );
     }
   }
+
+  protected async isFirstPartnership(projectId: string): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .match([
+        node('project', 'Project', { id: projectId }),
+        relation('out', '', 'partnership', { active: true }),
+        node('partnership'),
+      ])
+      .return(['partnership'])
+      .asResult<{ partnership?: Node }>()
+      .first();
+
+    return result?.partnership ? false : true;
+  }
+
+  /**
+   *
+   * match current primary partnership, its project, and all other partnerships
+   * set current to primary false
+   * set first of other partnerships to primary true
+   */
+  protected async movePrimaryPartnership(
+    value: boolean,
+    {
+      partnershipId,
+      projectId,
+    }: MergeExclusive<{ partnershipId: string }, { projectId: string }>
+  ): Promise<void> {
+    const query = this.db.query();
+    projectId
+      ? query.match([
+          node('project', 'Project', { id: projectId }),
+          relation('out', '', 'partnership', { active: true }),
+          node('partnership'),
+          relation('out', '', 'primary'),
+          node('property', 'Property', { value: true }),
+        ])
+      : query.match([
+          node('partnership', 'Partnership', { id: partnershipId }),
+        ]);
+
+    query
+      .optionalMatch([
+        node('partnership'),
+        relation('in', '', 'partnership', { active: true }),
+        node('project', 'Project'),
+        relation('out', '', 'partnership', { active: true }),
+        node('otherPartnership'),
+      ])
+      .optionalMatch([
+        node('partnership'),
+        relation('out', '', 'primary'),
+        node('property', 'Property'),
+      ])
+      .setValues({ 'property.value': false })
+      .with('project, otherPartnership')
+      .limit(1)
+      .optionalMatch([
+        node('otherPartnership'),
+        relation('out', '', 'primary'),
+        node('property', 'Property'),
+      ])
+      .setValues({ 'property.value': value });
```

Comment:
This is looking good, but we need to account for history as well. The current props need to be set to active=false and new ones created with active=true. I know this complicates the query more. See [`DatabaseService.updateProperty`](https://github.com/SeedCompany/cord-api-v3/blob/d1319a880a9ce7d5a66c93265ace25ee7ae350e5/src/core/database/database.service.ts#L306-L325) for example.